grapple/types/images.py

```diff
@@ -71,64 +109,80 @@ class ImageObjectType(DjangoObjectType):
     sizes = graphene.String(required=True)
     collection = graphene.Field(lambda: CollectionObjectType, required=True)
     tags = graphene.List(graphene.NonNull(lambda: TagObjectType), required=True)
-    rendition = graphene.Field(
-        lambda: get_rendition_type(),
-        max=graphene.String(),
-        min=graphene.String(),
-        width=graphene.Int(),
-        height=graphene.Int(),
-        fill=graphene.String(),
-        format=graphene.String(),
-        bgcolor=graphene.String(),
-        jpegquality=graphene.Int(),
-        webpquality=graphene.Int(),
-    )
+    rendition = graphene.Field(get_rendition_type, **get_rendition_field_kwargs())
     src_set = graphene.String(
         sizes=graphene.List(graphene.Int), format=graphene.String()
     )
+    if WAGTAIL_VERSION > (5, 0):
+        is_svg = graphene.Boolean(required=True)
 
     class Meta:
         model = WagtailImage
 
-    def resolve_rendition(instance, info, **kwargs):
+    def resolve_rendition(
+        instance: WagtailImage, info: GraphQLResolveInfo, **kwargs
+    ) -> WagtailImageRendition | None:
         """
         Render a custom rendition of the current image.
         """
-        filters = "|".join([f"{key}-{val}" for key, val in kwargs.items()])
+        preserve_svg = kwargs.pop("preserve_svg", False)
+        filter_specs = "|".join([f"{key}-{val}" for key, val in kwargs.items()])
+
+        # Only allow the defined filters (thus renditions)
+        if not rendition_allowed(filter_specs):
+            raise TypeError(
+                "Invalid filter specs. Check the `ALLOWED_IMAGE_FILTERS` setting."
+            )
+
+        if instance.is_svg() and preserve_svg:
+            # when dealing with SVGs, we want to limit the filter specs to those that are safe
+            filter_specs = to_svg_safe_spec(filter_specs)
 
-        # Only allowed the defined filters (thus renditions)
-        if not rendition_allowed(filters):
-            return
-        try:
-            return instance.get_rendition(filters)
-        except SourceImageIOError:
-            return
+            if not filter_specs:
+                raise TypeError(
+                    "No valid filter specs for SVG. "
+                    "See https://docs.wagtail.org/en/stable/topics/images.html#svg-images for details."
+                )
+
+        # previously we wrapped this in a try/except SourceImageIOError block.
+        # Removed to allow the error to bubble up in the response ("errors") and be handled by the user.
```

Comment:
This will go as a note in the changelog and release notes. I do want to keep it for a while as it is easier to have inline code context than scour release notes.